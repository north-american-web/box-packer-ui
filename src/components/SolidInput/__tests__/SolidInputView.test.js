import React from 'react';
import 'jest-dom/extend-expect';
import { render, fireEvent, cleanup } from '@testing-library/react';
import {SolidInputView} from '../index'

afterEach(cleanup);

describe('<SolidInputView/>', () => {
    it('renders and displays correctly with invalid data', () => {
       let isInputInvalid = true;
       const inputKey = 'test-key',
           inputValue = '11';

       const {getByLabelText, asFragment} = render(<SolidInputView
           isInputInvalid={isInputInvalid}
           inputKey={inputKey}
           onInputFieldChange={() => {}}
           onInputKeyPress={() => {}}
           onRemove={() => {}}
           inputValue={inputValue}
       />);

       const input = getByLabelText('Solid input');
       expect(input).toHaveAttribute('aria-invalid', "true");
       expect(input).toHaveAttribute('id', 'solid-input_test-key');
       expect(input).toHaveClass('is-error');
       expect(input).toHaveValue('11');

       expect(asFragment()).toMatchSnapshot();
    });

    it('doesn\'t use the error class if data is invalid and empty', () => {
        const {getByLabelText} = render(<SolidInputView
            isInputInvalid={true}
            inputKey='test-key'
            onInputFieldChange={() => {}}
            onInputKeyPress={() => {}}
            onRemove={() => {}}
            inputValue=''
        />);

        expect(getByLabelText('Solid input')).not.toHaveClass('is-error');
    });

    it('renders and displays correctly with valid data', () => {
        let isInputInvalid = false;
        const inputKey = 'test-key',
            inputValue = '1,1,1';

        const {getByLabelText} = render(<SolidInputView
            isInputInvalid={isInputInvalid}
            inputKey={inputKey}
            onInputFieldChange={() => {}}
            onInputKeyPress={() => {}}
            onRemove={() => {}}
            inputValue={inputValue}
        />);

        const input = getByLabelText('Solid input');
        expect(input).toHaveAttribute('aria-invalid', "false");
        expect(input).not.toHaveClass('is-error');
    });

    it('fires events correctly', () => {
        let isInputInvalid = true;
        const inputKey = 'test-key',
            onInputKeyPress = jest.fn(),
            onInputFieldChange = jest.fn(),
            onRemove = jest.fn(),
            inputValue = '1,1,1';

        const {getByLabelText} = render(<SolidInputView
            isInputInvalid={isInputInvalid}
            inputKey={inputKey}
            onInputFieldChange={onInputFieldChange}
            onInputKeyPress={onInputKeyPress}
            inputValue={inputValue}
            onRemove={onRemove}
        />);

        const input = getByLabelText('Solid input');

        fireEvent.click(getByLabelText('Delete item'));
        expect(onRemove.mock.calls.length).toBe(1);

        fireEvent.keyPress(input, {key: 'Enter', code:13, charCode: 13});
        expect(onInputKeyPress.mock.calls.length).toBe(1);

        fireEvent.change(input, { target: { value: 'a' }});
        expect(onInputFieldChange.mock.calls.length).toBe(1);
    })
});