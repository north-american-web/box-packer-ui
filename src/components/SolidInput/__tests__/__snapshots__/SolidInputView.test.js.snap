// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SolidInputView/> renders and displays correctly with invalid data 1`] = `
<DocumentFragment>
  <div
    class="form-group"
  >
    <div
      class="input-group"
    >
      <input
        aria-invalid="true"
        aria-label="Solid input"
        autocomplete="off"
        class="solid-input__input form-input is-error"
        id="solid-input_test-key"
        placeholder="width, length, height, description"
        type="text"
        value="11"
      />
      <button
        aria-label="Delete item"
        class="btn btn-default input-group-btn"
        type="button"
      >
        <svg
          fill="currentColor"
          height="16"
          size="16"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 448 512"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16zM53.2 467a48 48 0 0 0 47.9 45h245.8a48 48 0 0 0 47.9-45L416 128H32z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
